docs/models_dev_guide.md

```diff
@@ -0,0 +1,263 @@
+# Models Developer's Guide
+
+This document describes the design of the model management system in ETA and
+best practices for using it.
+
+
+## Motivation
+
+The ETA library relies on a number of large models in order to power its
+analytics. Models are trained on different machines, by different developers,
+and constantly need to be shared. We frequently generate new versions of a
+given model over time as we improve the implementation of the analytic and
+train it on evolving datasets. In addition, models are valuable, so they need
+to be archived and backed up to ensure that they remain secure and readily
+accessible.
+
+The ETA library provides a simple, yet powerful interface that satisifies
+these constraints.
+
+
+## Design
+
+The `eta.core.models` module implements the core interface for managing models
```

Comment:
Wondering if we want to be clear with our usage of "module" in ETA documentation.  This here is a "python module".

Reply:
I think its clear in this case because only Python modules use the `eta.core.XXX` syntax